atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -194,4 +223,58 @@ private Column makeColumn(long ts) {
     private static ByteBuffer getRowName() {
         return ByteBuffer.wrap(PtBytes.toBytes(ROW_AND_COLUMN_NAME));
     }
+
+    private String replaceBracesWithStringFormatSpecifier(String msg) {
+        return msg.replaceAll("\\{\\}", "%s");
+    }
+
+    private String getCurrentTimestampValues(CASResult result) {
+        return result.current_values.stream()
+                .map(Column::getValue)
+                .map(PtBytes::toString)
+                .collect(Collectors.joining(", "));
+    }
+
+    @VisibleForTesting
+    long getId() {
+        return id;
+    }
+
+    private static final class IdAndTimestamp {
+        private final Optional<Long> id;
+        private final long timestamp;
+
+        private IdAndTimestamp(byte[] values) {
+            String stringValue = PtBytes.toString(values);
+            Matcher columnStringMatcher = TIMESTAMP_FORMAT_PATTERN.matcher(stringValue);
```

Comment:
Very much of an edge case, but we run the risk of a wrong interpretation, if the byte representation of a long in the old format incidentally happens to match a string with one underscore somewhere in it. For example,

```java
PtBytes.toString(PtBytes.toBytes(3557616313682636848L)); // returns "1_100000"
```

Not sure if it's worth protecting against this case, but it's worth flagging that it's out there.